### DISCUSSION TITLE
How to compile and backtest EA in Linux without using Docker?

### DISCUSSION THREAD

**Discussion by codeitlikemiley** [EA31337/EA31337#296] [Q&A] [answered]
Hi i would like to ask if it is possible to do backtest/compile code without using docker?

May i know some of the requirements

just my hunch i think i will need wine to make this work.

I already have it , and a MT4 trader .exe software...

I dont wanna use docker since , it east up a lot of memory.

i always run out of memory since its only 8gb ram.

maybe using bare metal archlinux with all the required sofware can make it reduce the necessary ram to consume and use my gpu and processor at max capability.

Can you point me to the right direction thanks :)

**Comment by kenorb** [comment, marked as answer]
### Backtesting

Yes, please check the steps at [Backtesting in MT4](https://github.com/EA31337/EA31337/wiki/Backtesting-in-MT4) wiki page (_Manual method_). These steps should be very easy to follow. If something is not clear, please let me know.

Basically, you need to download already converted FXT/HST files into your platform folder, then run the test in MT4 platform as usual using a _Strategy Tester_ (the easiest way). Or you can download CSV files on your own (e.g. from the [FX-Data repos](https://github.com/FX-Data)), and convert it using a [`convert_csv_to_mt.py`](https://github.com/EA31337/FX-BT-Scripts) Python script into FXT/HST formats (more advanced and time consuming method).

For Linux, all you need is [`wine` command](https://www.winehq.org/), install the platform (download from one of the broker's websites) and run it.

For macOS, check [XM MT4 for Mac](https://www.xm.com/metatrader-mac) from XM broker.

### Compiling

To compile the project under Linux, the easiest way is to use `make` command. After cloning the project, run: `make Lite-Backtest`, or change `Lite-Backtest` into one of the following options found in `Makefile` file (such as `Lite`, `Advanced` or `Rider`).

To compile the project manually from the command-line, run `wine metaeditor.exe /compile:src`. For details, check: [How to compile an MQL4 file with a command-line tool?](https://stackoverflow.com/a/32459182/55075)
